.github/workflows/ghci.yml

```diff
@@ -0,0 +1,146 @@
+name: GitHub CI tests
+
+on:
+  push:
+    branches: [master, ci, coverity_scan]
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, macos-10.15]
+        cc: [gcc, clang]
+        exclude:
+          - os: macos-10.15
+            cc: gcc
+        include:
+          # Enable distcheck for one build
+          - os: ubuntu-18.04
+            cc: gcc
+            do_distc: yes
+          # Run Coverity on a clang build; Coverity's gcc causes issues
+          - os: ubuntu-18.04
+            cc: clang
+            do_coverity: yes
+
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2
+
+      - name: retrieve caches
+        uses: actions/cache@v2
+        with:
+          path: |
+            $HOME/Library/Caches/Homebrew
+            /usr/local/Homebrew
+          key: ${{ runner.os }}-${{ matrix.os }}-build
+
+      - name: install deps
+        run: |
+          if (echo "${{ matrix.os }}" | grep -qE '^macos-') ; then
+            #brew update
+            # freetype, autoconf and libtool are preinstalled
+            # and `brew install` fails if a non-uptodate version is already installed
+            #brew upgrade freetype autoconf libtool
+            brew install automake fribidi harfbuzz nasm
+          else
+            #sudo apt-get update && sudo apt-get upgrade
+            sudo apt-get install -y \
+                 libfontconfig1-dev libfreetype6-dev libfribidi-dev \
+                 libharfbuzz-dev nasm ${{ matrix.cc }}
+          fi
+
+      - name: configure
+        run: ./autogen.sh && CC="${{ matrix.cc }}" ./configure
+
+      - name: compile
+        run: make -j 2
```

Comment:
Does `make distcheck` require the regular compile step to be run beforehand? Could we save time (without sacrificing completeness or output clarity) by skipping this when distcheck is enabled?

Similarly, if we’re running Coverity Scan, it will recompile everything, so we could save time by skipping this step. However, we must be careful to fall back to plain compile if the Coverity Scan quota is exceeded.

Reply:
> Re: distcheck

It does not require a regular compile beforehand; currently we _could_ skip make if distcheck is enabled. However when we add regression-tests in the future, those will require a regular compile beforehand.

> Re: Coverity

As we already noticed, Coverity uses custom compiler builds which might change the build success. Imho it's better to also run a build with "real" clang.

We could also run coverity-only and  distcheck-only builds in parallel to regular builds. However, with the start, dependency-install and configure overhead, I'm not sure if that would actually be faster.